### PR TITLE
Merge pubspec.yaml and flutter.yaml.

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
Obviously this will require an e-mail and so forth.

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/2343

**Comment by abarth** [comment]
It might be valuable to have a period of time where the tool accepts both kinds.  For example, it could look at `pubspec.yaml` first and look for `flutter.yaml` if it doesn't find the `flutter` key.  (If it does find the `flutter` key, it should verify that `flutter.yaml` doesn't exist.

Also, consider a message in the case where it falls back to `flutter.yaml` explaining how to transition to the new location.

**Comment by Hixie** [comment]
Transitioning is really simple. How about just detecting the `flutter.yaml` file and saying what to do if it exists? (Rather than actually parsing both, which would require multiple schemas, etc.)

**Comment by abarth** [comment]
ok

**Comment by Hixie** [comment]
I have added a check for flutter.yaml files that gets triggered for the same commands that previously checked for a pubspec.yaml file, along with a multiparagraph precise step-by-step explanation for what to do and a pointer to an example. PTAL. :-)

**Comment by abarth** [comment]
LGTM

**Comment by Hixie** [comment]
I'll merge this while I'm at the office so that I can provide better support to anyone it breaks.

I'll post the following e-mail when I do so:

> **flutter.yaml merging into pubspec.yaml**
>
> The next time you update your Flutter repository, you will find that you need to make a change: you will need to merge your flutter.yaml file into your pubspec.yaml file and then delete your flutter.yaml file.
>
> Just running "flutter run" will give you detailed instructions on how to do this, or you can contact me and I'll help you out.
>
> This is part of an attempt to simplify the Flutter development experience by having fewer files to deal with.

**Comment by mit-mit** [comment]
Woot, happy to see this being fixed!

**Comment by mit-mit** [comment]
@devoncarew @pq do we use the presence of a flutter.yaml file for anything in IJ?

**Comment by pq** [comment]
@mit-mit: 👍 => https://github.com/flutter/flutter-intellij/issues/656

File: packages/flutter_tools/templates/create/pubspec.yaml.tmpl [hunk 1]
```diff
@@ -8,3 +9,37 @@ dev_dependencies:
...
+  # The following line ensures that the Material Icons font is
+  # included with your application, so that you can use the icons in
+  # the Icons class.
+  uses-material-design: true
+
+  # To add assets to your application, add an assets section here, in
+  # this "flutter" section, as in:
+  # assets:
+  #  - images/a_dot_burr.jpeg
+  #  - images/a_dot_ham.jpeg
```
Comment (mcornella) [inline packages/flutter_tools/templates/create/pubspec.yaml.tmpl]: To whomever did this, I see you 👀 
hamilton the musical scene

**Pull request merged**